### PR TITLE
[help.keyman.com] Add guide for creating lexical model package

### PR DESCRIPTION
**Pull request by darcywong00**
This PR adds a guide on creating a lexical model package based on the content from
https://help.keyman.com/developer/11.0/guides/distribute/tutorial/

Some of the steps are adjusted since the Keyman Developer 12.0 beta creates many of the package-related files when the lexical model project is created.

### CODE REVIEW

File: developer/12.0/guides/lexical-models/distribute/tutorial/step-1.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+<p>What is the purpose of a lexical model package? To make installation of a lexical model as
+straightforward as possible for the end user.</p>
+
+<p>We need to keep this goal in mind as we work on all the aspects of a package.</p>
+
+<p>A great package will:</p>
+
+<ol type="1">
+  <li>Include one lexical model to install.</li>
+  <li>Include a 'welcome.htm' file which is displayed before install.</li>
```
Comment (eddieantonio) [inline developer/12.0/guides/lexical-models/distribute/tutorial/step-1.php]: Is displaying the `welcome.htm` supported on iOS and Android?

File: developer/12.0/guides/lexical-models/distribute/tutorial/step-2.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      <p>The <code class='language-none'>link:</code> sceheme will open the referred file in the default application - that is, a web browser for 
+      URLs and links, Notepad for .txt files, Adobe Reader for PDFs. You can use <code class='language-none'>link:</code> to open any of the files in 
+      the package, e.g. <code class='language-none'>link:docs.pdf</code> will open the file docs.pdf in Adobe Reader or the default PDF reader on the
+      system.</p>
+    </aside>
+
+    <p>If you create documents in other formats, for example PDF or printable documentation, you should link to that in the
+    welcome.htm.</p>
+
+    <!-- TODO: Is this still valid?
```
Comment (eddieantonio) [inline developer/12.0/guides/lexical-models/distribute/tutorial/step-2.php]: It is no longer valid. See keymanapp/keyman#186

Please delete this commented-out prose.

File: developer/12.0/guides/lexical-models/distribute/tutorial/step-2.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  </dd>
+
+  <dt>Lexical Model</dt>
+
+  <dd>
+    <p>Add <strong>one</strong> .model.js compiled Keyman lexical model to the package. While keyboards can also be distributed in
+      packages, do not include them in a lexical model package.</p>
+  </dd>
+
+  <dt>Documentation</dt>
```
Comment (eddieantonio) [inline developer/12.0/guides/lexical-models/distribute/tutorial/step-2.php]: How relevant is this section to lexical model packages? Do we have a way to display this (unspecified) documentation in the apps? We don't want package authors writing documentation that will be inaccessible to end users.

File: developer/12.0/guides/lexical-models/distribute/tutorial/step-5.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  <li>Include a link to the Keyman download page: http://keyman.com/downloads/</li>
+</ol>
+<h2>Distributing a package by email</h2>
+<li>Attaching the KMP file directly to an email may be blocked for security reasons. As mentioned above, a KMP file is
+  basically a ZIP file and lexical model data is in JavaScript, this is a combination that looks suspicious to many
+  email servers. You can upload it to a Google drive, and email a link for downloading the file</li>
+
+
+<li>You may need to do some exploring to get your mobile device to load the lexical model from the KMP file into Keyman.
```
Comment (eddieantonio) [inline developer/12.0/guides/lexical-models/distribute/tutorial/step-5.php]: Is this still up-to-date?

File: developer/12.0/guides/lexical-models/distribute/tutorial/step-1.php (hunk 1 above)
Comment (darcywong00) [reply developer/12.0/guides/lexical-models/distribute/tutorial/step-1.php]: Yes. It follows the pattern from keyboard .kmp installation where the mobile apps display `welcome.htm` as a place for users to confirm installation or cancel.

File: developer/12.0/guides/lexical-models/distribute/tutorial/step-5.php (hunk 4 above)
Comment (darcywong00) [reply developer/12.0/guides/lexical-models/distribute/tutorial/step-5.php]: Maybe the download is only applicable for Android?
Keyman 12.0 for Android adds a file-browser from the in-app Settings panel for the user to browse to install.

@jahorton does iOS only trigger KMP installs from URL links and email attachments?

File: developer/12.0/guides/lexical-models/distribute/tutorial/step-2.php (hunk 3 above)
Comment (darcywong00) [reply developer/12.0/guides/lexical-models/distribute/tutorial/step-2.php]: I'll remove the extra Documentation section and only have authors update readme.htm and welcome.htm.

File: developer/12.0/guides/lexical-models/distribute/tutorial/step-5.php (hunk 4 above)
Comment (darcywong00) [reply developer/12.0/guides/lexical-models/distribute/tutorial/step-5.php]: Cross-referenced pages for installing custom keyboard packages

File: developer/12.0/guides/lexical-models/distribute/tutorial/step-3.php [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  You can add multiple files at once, and from multiple folders. When the package is compiled, all the files will be
+  placed in the same folder within the package.</p>
+
+<p>While keyboards can also be distributed in packages, do not include them in a lexical model package.</p>
+
+<h2>Lexical Models</h2>
+<p>In the Package Editor, click on the <span class="guibutton">Lexical Models</span> tab.</p>
+
+<p><img src="<?php echo cdn("img/developer/120/lm/tutorial_distribute_model_3.png"); ?>"
+        alt="Lexical Model Info"/></p>
```
Comment (darcywong00) [inline developer/12.0/guides/lexical-models/distribute/tutorial/step-3.php]: Note: the screenshot will be updated later to reflect the Developer change of moving "Version" to the "Details" tab

File: developer/12.0/guides/lexical-models/distribute/tutorial/step-5.php (hunk 4 above)
Comment (jahorton) [reply developer/12.0/guides/lexical-models/distribute/tutorial/step-5.php]: > 
> 
> Maybe the download is only applicable for Android?
> Keyman 12.0 for Android adds a file-browser from the in-app Settings panel for the user to browse to install.
> 
> @jahorton does iOS only trigger KMP installs from URL links and email attachments?

iOS doesn't provide Finder like macOS does.  While there _are_ apps that will provide similar functionality, it's not something iOS provides by default that Keyman can access to independently open files.  So, to the best of my knowledge, this is correct.

Comment (jahorton) [reply developer/12.0/guides/lexical-models/distribute/tutorial/step-5.php]: For a bit extra on that, we have this longstanding issue on the main repo:  https://github.com/keymanapp/keyman/issues/1088

We don't currently support installation via iTunes, even.  The supported ways to get a .kmp for install on iOS are fairly limited.

**Pull request merged**